share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml

```diff
@@ -884,10 +889,50 @@ deprecated-ci-build:
     when:
       - always
 
+.generate-cray-sles:
```

Comment:
This section seems to duplicate a lot of the main generate script.  Is it possible #38514 (when merged) could help avoid the need for that?  Identifying the specific things that must be different (e.g. commands that can't, or must, be run on cray) could help here.

Reply:
The only difference right now between `generate-cray-sles` and `generate-cray` is the tagging, I think. More changes may be necessary as we flesh this out but for now that was the immediate motivation for duplicating the generate job. If there is a way to get the tagging right without duplicating the other elements, I'm in favor of that!

Reply:
Just pushed a commit to reduce duplication per your comment:
https://github.com/spack/spack/pull/38744/commits/67e025d5be6d385c0f44ea2c2880f97a7ab2aab4

What do you think?